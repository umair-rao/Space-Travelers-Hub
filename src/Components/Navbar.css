* {
  padding: 0;
  margin: 0;
}

.navbar {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
  height: 5rem;
  width: 100%;
  margin-bottom: 2rem;
  border-bottom: 1px solid rgb(204, 204, 204);
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.logo {
  display: flex;
  align-items: center;
  margin-left: 2rem;
}

.logo h1 {
  font-size: 2rem;
}

.logo-image {
  height: 5rem;
  padding-right: 3rem;
}

.nav-links {
  display: flex;
  align-items: center;
  margin-right: 3rem;
}

.nav-links li {
  display: inline;
  list-style-type: none;
  padding: 0.05rem 0;
  text-align: center;
  margin: 0 1rem;
}

.link {
  text-decoration: none;
  color: rgb(52, 129, 154);
  font-size: 1.3rem;
}

.link.active {
  text-decoration: underline;
}