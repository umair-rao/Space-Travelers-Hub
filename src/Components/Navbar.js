import React from 'react';
import { Link, useLocation } from 'react-router-dom';
import logo from '../logo/planet.png';
import './Navbar.css';

const Navbar = () => {
  const location = useLocation();
  const { pathname } = location;

  return (
    <nav className="navbar">
      <Link to="/" className="logo">
        <img className="logo-image" src={logo} alt="Logo" />
        <h1>Space Travelers` Hub</h1>
      </Link>
      <div className="nav-links">
        <ul>
          <li><Link to="/" className={`link ${pathname === '/' ? 'active' : ''}`}>Rockets</Link></li>
          <li><Link to="/missions" className={`link ${pathname === '/missions' ? 'active' : ''}`}>Missions</Link></li>
          <li><Link to="/dragons" className={`link ${pathname === '/dragons' ? 'active' : ''}`}>Dragons</Link></li>
          <li><Link to="/profile" className={`link ${pathname === '/profile' ? 'active' : ''}`}>Profile</Link></li>
        </ul>
      </div>
    </nav>
  );
};

export default Navbar;
