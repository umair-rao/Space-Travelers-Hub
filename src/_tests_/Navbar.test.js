import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Components/Navbar';

describe('Navbar tests', () => {
  test('navbar render', () => {
    const navbar = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(navbar).toMatchSnapshot();
  });
});
