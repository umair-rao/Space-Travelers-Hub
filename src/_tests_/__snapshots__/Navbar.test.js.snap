// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar tests navbar render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar"
      >
        <a
          class="logo"
          href="/"
          style="text-decoration: none;"
        >
          <img
            alt="Logo"
            class="logo-image"
            src="planet.png"
          />
          <h1
            class="nav-heading"
          >
            Space Travelers\` Hub
          </h1>
        </a>
        <div
          class="nav-links"
        >
          <ul>
            <li>
              <a
                class="link active"
                href="/"
              >
                Rockets
              </a>
            </li>
            <li>
              <a
                class="link "
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li>
              <a
                class="link "
                href="/dragons"
              >
                Dragons
              </a>
            </li>
            <li>
              <a
                class="link "
                href="/profile"
              >
                Profile
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar"
    >
      <a
        class="logo"
        href="/"
        style="text-decoration: none;"
      >
        <img
          alt="Logo"
          class="logo-image"
          src="planet.png"
        />
        <h1
          class="nav-heading"
        >
          Space Travelers\` Hub
        </h1>
      </a>
      <div
        class="nav-links"
      >
        <ul>
          <li>
            <a
              class="link active"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              class="link "
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              class="link "
              href="/dragons"
            >
              Dragons
            </a>
          </li>
          <li>
            <a
              class="link "
              href="/profile"
            >
              Profile
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
